### PR TITLE
feat(dedicated): adding banner for autorenew 2016 deployment

### PR DESCRIPTION
**Pull request by JacquesLarique**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11781, MANAGER-11782
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~~[ ] Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits (n/a)

## Description

Adding a banner for autorenew 2016 deployment

## Related
- [x] translations: CMT-1596

### CODE REVIEW

File: packages/manager/modules/billing/src/autoRenew/autorenew.routing.js [hunk 1]
```diff
@@ -89,13 +89,20 @@ export default /* @ngInject */ ($stateProvider, coreConfigProvider) => {
...
-          ovhFeatureFlipping
-            .checkFeatureAvailability(['billing:management'])
-            .then((commitmentAvailability) =>
-              commitmentAvailability.isFeatureAvailable('billing:management'),
-            )
-            .catch(() => false),
+        featureAvailability: /* @ngInject */ (ovhFeatureFlipping) =>
+          ovhFeatureFlipping.checkFeatureAvailability([
+            'billing:management',
+            'billing:autorenew2016Deployment',
```
Comment (sachinrameshn) [inline packages/manager/modules/billing/src/autoRenew/autorenew.routing.js:95]: Suggestion: Can you move this to constant ?

**Review by anooparveti — APPROVED** [review]

File: packages/manager/modules/billing/src/autoRenew/autorenew.routing.js [hunk 2]
```diff
@@ -89,13 +89,20 @@ export default /* @ngInject */ ($stateProvider, coreConfigProvider) => {
...
+          ]),
+        isAutorenewManagementAvailable: /* @ngInject */ (featureAvailability) =>
+          featureAvailability?.isFeatureAvailable('billing:management') ||
+          false,
+        isAutorenew2016DeploymentBannerAvailable: /* @ngInject */ (
+          featureAvailability,
+        ) =>
+          featureAvailability?.isFeatureAvailable(
+            'billing:autorenew2016Deployment',
```
Comment (sachinrameshn) [inline packages/manager/modules/billing/src/autoRenew/autorenew.routing.js:104]: Suggestion: Can you move this to constant ?

**Review by sachinrameshn — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

File: packages/manager/modules/billing-components/src/components/auto-renew-2016-deployment-banner/auto-renew-2016-deployment-banner.controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export default class AutoRenew2016DeploymentBannerController {
+  /* @ngInject */
+  constructor(coreConfig) {
+    this.coreConfig = coreConfig;
+  }
+
+  $onInit() {
+    this.FAQ_LINK =
+      FAQ_LINK[this.coreConfig.getUserLanguage().toUpperCase()] ||
+      FAQ_LINK[this.coreConfig.getUser().ovhSubsidiary] ||
```
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/auto-renew-2016-deployment-banner/auto-renew-2016-deployment-banner.controller.js:12]: **nitpick:** Guides must always be displayed based on Subsidiary. Use only has the option to change the language in Manager.
```suggestion
      FAQ_LINK[this.coreConfig.getUser().ovhSubsidiary] ||
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/billing-components/src/components/auto-renew-2016-deployment-banner/auto-renew-2016-deployment-banner.controller.js (hunk 3 above)
Comment (qpavy) [reply packages/manager/modules/billing-components/src/components/auto-renew-2016-deployment-banner/auto-renew-2016-deployment-banner.controller.js:12]: I added subsidiaries as the guide links were only display based on language but I can remove the retrieve with user language as well

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**